spyql/parser.py

```diff
@@ -72,8 +72,16 @@ def run(self, keyword: Optional[str]):
 
 
 def parse_structure(query: str):
-    tokens = re.split(r" |\n", query.strip())
+    tokens: List[str] = []
+    for line in query.split("\n"):
+        if "#" in line:
+            # Remove comment from parse targets
+            line = line.split("#")[0]
+
+        tokens += line.split(" ")
```

Comment:
We should have a way to escape the `#` in case the user wants to use it somewhere in the query. Maybe we should check if it's part of some string and don't split the line in that case.

Reply:
One note: at this point, strings were already replaced by placeholders. e.g. 

```sql 
SELECT "I prefer C# to Java" AS txt`
``` 

would be rendered as something like 

```sql
SELECT __jgh59ah40myd__ AS txt
```

@recharte, besides strings, I am not seeing any other case where we would use `#`, I am missing something?

Reply:
Just to be sure, I added a test to see if '#' in the string is ignored

Reply:
You're right Daniel, I totally forgot that the string has already been processed. Please disregard my comment.